src/subdomains/generic/kyc/enums/kyc.enum.ts

```diff
@@ -17,24 +26,21 @@ export function getKycStepIndex(stepName: KycStepName): number {
 }
 
 export function requiredKycSteps(userData: UserData): KycStepName[] {
-  return userData.accountType === AccountType.BUSINESS
-    ? [
-        KycStepName.CONTACT_DATA,
-        KycStepName.PERSONAL_DATA,
-        KycStepName.NATIONALITY_DATA,
-        KycStepName.COMMERCIAL_REGISTER,
-        KycStepName.IDENT,
-        KycStepName.FINANCIAL_DATA,
-        KycStepName.DFX_APPROVAL,
-      ]
-    : [
-        KycStepName.CONTACT_DATA,
-        KycStepName.PERSONAL_DATA,
-        KycStepName.NATIONALITY_DATA,
-        KycStepName.IDENT,
-        KycStepName.FINANCIAL_DATA,
-        KycStepName.DFX_APPROVAL,
-      ];
+  return [
+    KycStepName.CONTACT_DATA,
+    KycStepName.PERSONAL_DATA,
+    userData.accountType === AccountType.BUSINESS ? KycStepName.LEGAL_ENTITY : null,
+    userData.legalEntity === LegalEntity.PUBLIC_LIMITED_COMPANY ? KycStepName.STOCK_REGISTER : null,
+    KycStepName.NATIONALITY_DATA,
+    [AccountType.BUSINESS, AccountType.BUSINESS].includes(userData.accountType)
```

Comment:
Do you mean `AccountType.SOLE_PROPRIETORSHIP`?

Reply:
Ok but I mean the `[AccountType.BUSINESS, AccountType.BUSINESS]`, so two time the same value in the array makes no sense I think, or? 😅